examples/basics/plotting/grid_x_y_viewbox.py

```diff
@@ -0,0 +1,48 @@
+
+
+# -*- coding: utf-8 -*-
+# vispy: gallery 30
+# -----------------------------------------------------------------------------
+# Copyright (c) 2015, Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+
+"""
++---+-------------+
+|   |             |
+| y |   viewbox   |
+|   |             |
++---+-------------+
+| sp|     x       |
```

Comment:
One option is to set the min/max size constraints to be the level just below mandatory. Do you think that would work? They would need to be the only things on that level, though, so not sure.

Actually, can you add to the docstring somewhere which level each constraint is placed? It's important to say it somewhere for the next person who goes to make changes.